### DISCUSSION TITLE
G-code script for Cura / Prusa different

### DISCUSSION THREAD

**Discussion by codegain** [mriscoc/Ender3V2S1#1219] [Q&A]
Hi,

I recently switched to Prusa from Cura and copied the start scripts for Prusa from here:  https://github.com/mriscoc/Ender3V2S1/wiki/Slicer-G-code-Scripts

I noticed that there are some differences between the Cura and Prusa start script (not the variables or the order) in printing and later in the code (I didn't pay much attention at first).

For example:

For Cura:
```
...
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish
```

For Prusa:

```
...
G1 Z2 F240  ; F value is different from Cura
G1 X2 Y10 F3000  ; X/Y/F value is different from Cura - causes Prusa to draw the line further in the bed
G1 Z0.28 F240 ; Seems OK, but also different from Cura
G92 E0 ; Extruder not resetted in Cura
G1 Y190 E15 F1500 ; intro line - doesn't got to Y200 like Cura
G1 X2.3 F5000 ; seems fine
G92 E0 ; not in Cura script
G1 Y10 E15 F1200 ; intro line - different F value
G92 E0 ; "Reset extruder" in Cura script

; all other lines after "Reset extruder" from Cura script missing
```

In a print sliced by Prusa, the printhead does *not* raise after drawing the intro line and moves fast with Z0.3 to the middle of the printbed.

Is there a reason why theese (move) differences exist between these scripts or was the Prusa script just not updated when the Cura script got updated?
Is it safe to copy the "move" commands from Cura and use them in Prusa. I really liked that the intro line is further away and lifting of the nozzle before moving to the middle to prevent scratching.

**Comment by mriscoc** [comment]
The G-code scripts are based on the standard scripts in old Cura and in Prusaslicer versions so maybe they need an update but currently they work fine.
